crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -376,6 +376,32 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Pa
             .await
             .transpose()?;
 
+        let mandate_id = if mandate_id.is_none() {
+            request
```

Comment:
can you move this match to the place where it is previously being matched? Move this to a function maybe?

Reply:
Marking this as resolved as per our discussion.